content/src/main/content/jcr_root/apps/acs-samples/angularjs-integration/clientlibs/footerLib/js/angular/core/main.js

```diff
@@ -0,0 +1,30 @@
+/*global angular*/
+
+/*
+ * This is the definition of the application module itself as well as core configurations like routing and the http service.
+ * Since this file defines that module that all of our controllers and services are part of, it needs to be listed first in the clientlibs js.txt file.
+ */
+var sampleModule = angular.module('angularjs-sample', ['ngRoute']);
+
+/*
+ * The routeProvider configuration maps URL paths to AJAX content to be loaded into the ng-view container defined in the containerPage component.
+ * Note that the templateUrls are each using a .content Sling selector.  This allows us to switch out the content portion of the containerPage with that of the partials.
+ */
+sampleModule.config(['$routeProvider', function ($routeProvider) {
+        $routeProvider.
+                when('/home', {
+                    templateUrl: '/content/acs-samples/angularjs-integration/home.content.html',
```

Comment:
That's a good question.  I guess you could build out some tool to let an
author configure these and load them in via a call back to the repo or an
OSGi service.  It would probably add a fair amount of complexity here,
though.  In our actual implementation, we took this approach since there
was a predefined booking funnel that was not likely to change without the
need for developer intervention anyway.

On Sun, Mar 8, 2015 at 8:46 AM, david g. <notifications@github.com> wrote:

> In
> content/src/main/content/jcr_root/apps/acs-samples/angularjs-integration/clientlibs/footerLib/js/angular/core/main.js
> <https://github.com/Adobe-Consulting-Services/acs-aem-samples/pull/26#discussion_r26006206>
> :
>
> > +/*global angular*/
> > +
> > +/*
> > + * This is the definition of the application module itself as well as core configurations like routing and the http service.
> > + * Since this file defines that module that all of our controllers and services are part of, it needs to be listed first in the clientlibs js.txt file.
> > + */
> > +var sampleModule = angular.module('angularjs-sample', ['ngRoute']);
> > +
> > +/*
> > + * The routeProvider configuration maps URL paths to AJAX content to be loaded into the ng-view container defined in the containerPage component.
> > + * Note that the templateUrls are each using a .content Sling selector.  This allows us to switch out the content portion of the containerPage with that of the partials.
> > + */
> > +sampleModule.config(['$routeProvider', function ($routeProvider) {
> > +        $routeProvider.
> > +                when('/home', {
> > +                    templateUrl: '/content/acs-samples/angularjs-integration/home.content.html',
>
> Is it best practice to hardcode these URIs? Is there a simple, clean way
> around it?
>
> At the same time dont want to add too much complexity and orchestration
> that would distract from the fundamentals of the integration.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Adobe-Consulting-Services/acs-aem-samples/pull/26/files#r26006206>
> .
>